### PR TITLE
Fix ms catalog provider docs

### PR DESCRIPTION
**Pull request by scottmelhop**
Update configuration entry based on this:
image

image

Update Provider suggestion based on this:
image

### CODE REVIEW

**Comment by benjdlambert** [comment]
Hey @scottmelhop! Thanks for the change, this also will need a changeset so that we publish the readme to npm on next release :pray:

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thanks! 👍 

The existing documentation is actually correct as it. The provider is the new and recommended way to ingest org data, the processor is deprecated and will be removed. There should be a deprecation warning about the initial message that tells you to add config to `catalog.providers.microsoftGraphOrg` instead, and points you to docs.

**Comment by scottmelhop** [comment]
> The existing documentation is actually correct as it. The provider is the new and recommended way to ingest org data, the processor is deprecated and will be removed.

Ok, I just started trying to use it so not totally sure what you mean. Basically its very hard to follow at the moment, so not sure how the documentation should be updated

**Comment by scottmelhop** [comment]
How does the provider supposed to get the correct configuration?

**Comment by scottmelhop** [comment]
I still get this error following those docs
```
Backend failed to start up, Error: There is no Microsoft Graph Org provider that matches "undefined". Please add a configuration entry for it under "catalog.processors.microsoftGraphOrg.providers".
```

**Comment by scottmelhop** [comment]
So in `node_modules/@backstage/plugin-catalog-backend-module-msgraph/dist/index.cjs.js` I see this

```
static fromConfig(configRoot, options) {
    const config = configRoot.getOptionalConfig("catalog.processors.microsoftGraphOrg");
    const providers = config ? readMicrosoftGraphConfig(config) : [];
    const provider = providers.find((p) => options.target.startsWith(p.target));
    if (!provider) {
      throw new Error(`There is no Microsoft Graph Org provider that matches "${options.target}". Please add a configuration entry for it under "catalog.processors.microsoftGraphOrg.providers".`);
    }
```

Im far from a react/js expert, but it is expecting the `app-config.yaml` to have `catalog.processors.microsoftGraphOrg.providers` and not `catalog.providers.microsoftGraphOrg.default`

Am I missing something here?
